compiler/rustc_mir_transform/src/instsimplify.rs

```diff
@@ -140,6 +150,19 @@ impl<'tcx> InstSimplifyContext<'tcx, '_> {
         }
     }
 
+    fn simplify_ub_check(
+        &self,
+        source_info: &SourceInfo,
+        rvalue: &mut Rvalue<'tcx>,
+        debug_assertions: bool,
+    ) {
+        if let Rvalue::NullaryOp(NullOp::UbCheck(_), _) = *rvalue {
+            let const_ = Const::from_bool(self.tcx, debug_assertions);
```

Comment:
This transform breaks the documented semantics of these intrinsics in the interpreter: that UB checks are always disabled there.